src/main.ts

```diff
@@ -0,0 +1,547 @@
+  RUST_BACKTRACE?: string;
+  NIX_INSTALLER_MODIFY_PROFILE?: string;
+  NIX_INSTALLER_NIX_BUILD_GROUP_NAME?: string;
+  NIX_INSTALLER_NIX_BUILD_GROUP_ID?: string;
+  NIX_INSTALLER_NIX_BUILD_USER_PREFIX?: string;
+  NIX_INSTALLER_NIX_BUILD_USER_COUNT?: string;
+  NIX_INSTALLER_NIX_BUILD_USER_ID_BASE?: string;
+  NIX_INSTALLER_NIX_PACKAGE_URL?: string;
+  NIX_INSTALLER_PROXY?: string;
+  NIX_INSTALLER_SSL_CERT_FILE?: string;
+  NIX_INSTALLER_DIAGNOSTIC_ENDPOINT?: string;
+  NIX_INSTALLER_ENCRYPT?: string;
+  NIX_INSTALLER_CASE_SENSITIVE?: string;
+  NIX_INSTALLER_VOLUME_LABEL?: string;
+  NIX_INSTALLER_ROOT_DISK?: string;
+  NIX_INSTALLER_INIT?: string;
+  NIX_INSTALLER_START_DAEMON?: string;
+  NIX_INSTALLER_NO_CONFIRM?: string;
+  NIX_INSTALLER_EXTRA_CONF?: string;
+  NIX_INSTALLER_LOG_DIRECTIVES?: string;
+  NIX_INSTALLER_LOGGER?: string;
+};
+
+function get_nix_platform(): string {
+  const env_os = process.env.RUNNER_OS;
+  const env_arch = process.env.RUNNER_ARCH;
+
+  if (env_os === "macOS" && env_arch === "X64") {
+    return "x86_64-darwin";
+  } else if (env_os === "macOS" && env_arch === "ARM64") {
+    return "aarch64-darwin";
+  } else if (env_os === "Linux" && env_arch === "X64") {
+    return "x86_64-linux";
+  } else if (env_os === "Linux" && env_arch === "ARM64") {
+    return "aarch64-linux";
+  } else {
+    throw new Error(
+      `Unsupported \`RUNNER_OS\` (currently \`${env_os}\`) and \`RUNNER_ARCH\` (currently \`${env_arch}\`)  combination`,
+    );
+  }
+}
+
+function get_default_planner(): string {
+  const env_os = process.env.RUNNER_OS;
+
+  if (env_os === "macOS") {
+    return "macos";
+  } else if (env_os === "Linux") {
+    return "linux";
+  } else {
+    throw new Error(`Unsupported \`RUNNER_OS\` (currently \`${env_os}\`)`);
+  }
+}
+
+function resolve_nix_installer_url(platform: string): URL {
+  // Only one of these are allowed.
+  const nix_installer_branch = action_input_string_or_null(
+    "nix-installer-branch",
+  );
+  const nix_installer_pr = action_input_number_or_null("nix-installer-pr");
+  const nix_installer_revision = action_input_string_or_null(
+    "nix-installer-revision",
+  );
+  const nix_installer_tag = action_input_string_or_null("nix-installer-tag");
+  const nix_installer_url = action_input_string_or_null("nix-installer-url");
+
+  let num_set = 0;
+  if (nix_installer_branch !== null) {
+    num_set += 1;
+  }
+  if (nix_installer_pr !== null) {
+    num_set += 1;
+  }
+  if (nix_installer_revision !== null) {
+    num_set += 1;
+  }
+  if (nix_installer_tag !== null) {
+    num_set += 1;
+  }
+  if (nix_installer_url !== null) {
+    num_set += 1;
+  }
+  if (num_set > 1) {
+    throw new Error(
+      `The following options are mututally exclusive, but ${num_set} were set: \`nix_installer_branch\`, \`nix_installer_pr\`, \`nix_installer_revision\`, \`nix_installer_tag\`, and \`nix_installer_url\``,
+    );
+  }
+
+  if (nix_installer_branch !== null) {
```

Comment:
Very minor nit: I wonder if it might be a little nicer to set the URL we'll be using in the checks above, and then `return that_url` after we've checked that only 1 of those options is set.

Like, `let url = "default install url"; if branch != null, url = "branch install url";` or something.